### PR TITLE
Release 2.2.0

### PR DESCRIPTION
**Pull request by fillup**
- Remove unused output that causes errors in ecs service modules
- Add new public only VPC module
- Add support for additional security groups in autoscaling group
- fix list concat in asg module
- add support for public ips to asg
- add Name tag to asg instances
- allow publicly accessible db
- update ecs_service to use ordered_placement_strategy

### CODE REVIEW

File: aws/vpc-public-only/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+ - Create VPC named after `app_name` and `app_env`
+ - Create public subnet for each `aws_zones` specified
+ - Provision a Internet Gateway and configure public subnets to route through it
+ - Create a DB subnet group including all public subnets
+
+## Required Inputs
+
+ - `app_name` - Name of application, ex: Doorman, IdP, etc.
+ - `app_env` - Name of environment, ex: production, testing, etc.
```
Comment (forevermatt) [inline aws/vpc-public-only/README.md]: Do you want to shorten the suggested `app_env` values? (e.g. "prod, test, etc.")

File: aws/vpc-public-only/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ - `aws_zones` - A list of zones to create subnets in, ex: `["us-east-1c", "us-east-1d", "us-east-1e"]`
+
+## Optional Inputs
+
+ - `enable_dns_hostnames` - default `false`
```
Comment (forevermatt) [inline aws/vpc-public-only/README.md]: This says the default is `false`, but the code uses a default of `true`

File: aws/vpc-public-only/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ - Create VPC named after `app_name` and `app_env`
+ - Create public subnet for each `aws_zones` specified
+ - Provision a Internet Gateway and configure public subnets to route through it
+ - Create a DB subnet group including all public subnets
+
+## Required Inputs
+
+ - `app_name` - Name of application, ex: Doorman, IdP, etc.
+ - `app_env` - Name of environment, ex: production, testing, etc.
+ - `aws_zones` - A list of zones to create subnets in, ex: `["us-east-1c", "us-east-1d", "us-east-1e"]`
```
Comment (Baggerone) [inline aws/vpc-public-only/README.md]: Do you want to remove the default from this in the vars.tf? (Or move this description out of the "Required Inputs"?)

**Review by Baggerone — DISMISSED** [review]

File: aws/vpc-public-only/main.tf [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+/*
+ * Get default security group for reference later
+ */
+data "aws_security_group" "vpc_default_sg" {
+  name   = "default"
+  vpc_id = "${aws_vpc.vpc.id}"
+}
+
+/*
+ * Create public and private subnets for each availability zone
```
Comment (forevermatt) [inline aws/vpc-public-only/main.tf]: Does this actually create both public and private subnets, or just public subnets?

File: aws/vpc-public-only/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Default Security Group ID
```
Comment (forevermatt) [inline aws/vpc-public-only/outputs.tf]: This comment seems to have become disconnected from the code it's describing.

**Review by forevermatt — COMMENTED** [review]
Added a few readme/comment related comments, but otherwise this looks fine to me.

**Comment by dalenewby** [comment]
Variable associate_public_ip_address is missing from asg/README.md.

File: aws/vpc-public-only/README.md (hunk 1 above)
Comment (fillup) [reply aws/vpc-public-only/README.md]: sure

File: aws/vpc-public-only/README.md (hunk 3 above)
Comment (fillup) [reply aws/vpc-public-only/README.md]: yeah I'll remove defaults

File: aws/vpc-public-only/main.tf (hunk 4 above)
Comment (fillup) [reply aws/vpc-public-only/main.tf]: public only, thanks

File: aws/vpc-public-only/outputs.tf (hunk 5 above)
Comment (fillup) [reply aws/vpc-public-only/outputs.tf]: you're right, thanks

**Comment by dalenewby** [comment]
The example environment names are different in the README.md files (alb/, asg/, vpc/, ecs/cluster/, rds/mariadb/ vs. vpc-public-only/).

File: aws/asg/README.md [hunk 6]
```diff
@@ -21,13 +21,12 @@ an auto scaling group that uses the configuration.
 ## Optional Inputs
```
Comment (dalenewby) [inline aws/asg/README.md:21]: Should optional inputs be included in the "Example Usage" section?

**Comment by dalenewby** [comment]
The README.md for aws/ecs/service-only/ and aws/ecs/service-only-with-volume say:
- Create IAM Role: `ecsServiceRole`

Do they?  I don't see it in the code.

**Comment by dalenewby** [comment]
aws/ecs/service-with-alb/ doesn't have a README.md.

**Comment by dalenewby** [comment]
aws/rds/mariadb/README.md is missing optional inputs publicly_accessible, skip_final_snapshot, and parameter_group_name.

**Review by dalenewby — DISMISSED** [review]
I left some comments on the README.md files.  Otherwise, it looks ok to me.

**Review by forevermatt — DISMISSED** [review]

File: aws/asg/README.md (hunk 6 above)
Comment (fillup) [reply aws/asg/README.md:21]: eh, I usually leave them out for the simplest usage example

**Comment by fillup** [comment]
Thanks @dalenewby and @forevermatt for all the feedback, its all been updated.

**Pull request merged**